examples/resources/working_with_trait_versions/example.yml

```diff
@@ -0,0 +1,89 @@
+# yaml-language-server: $schema=https://raw.githubusercontent.com/OpenAssetIO/OpenAssetIO-TraitGen/main/python/openassetio_traitgen/schema.json
+# yamllint disable-line rule:document-start
+package: openassetio-example
+description: An example traits schema
+
+traits:
+  example:
+    description: Example namespace
+    members:
+
+      Deprecated:
+        deprecated: true
+        versions:
+          - description: An example.
+            usage:
+            - entity
+            - locale
+            - relationship
+
+      Added:
+        versions:
+          - description: An example.
+            usage:
+              - entity
+              - locale
+              - relationship
+
+      Updated:
+        versions:
+          - description: An example.
```

Comment:
[question] Do we need to explicitly version the traits/specifications in the YAML? So that we can delete them in the future without resetting the index? You'd mentioned a concern before for an ever growing list. If they were versioned you could remove deprecated ones after some period of time.